### PR TITLE
test(sec-core): add prompt-scanner e2e test suite and Makefile target

### PR DESCRIPTION
**Pull request by haosanzi**
- Add comprehensive e2e tests for prompt-scanner CLI via scan-prompt command
- Cover basic functionality (safe/unsafe prompts, injection, jailbreak detection)
- Add parametrized rule coverage tests for 20+ injection and jailbreak patterns
- Test mode variants (fast/standard/strict) and JSON output format validation
- Add error handling tests for invalid modes, formats, and edge cases
- Add Makefile target test-prompt-scanner-e2e with model warmup support

## Related Issue

closes #

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional change)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Scope

- [ ] `cosh` (copilot-shell)
- [ ] `sec-core` (agent-sec-core)
- [ ] `skill` (os-skills)
- [ ] `sight` (agentsight)
- [ ] `tokenless` (tokenless)
- [ ] Multiple / Project-wide

## Checklist

- [ ] I have read the [Contributing Guide](../CONTRIBUTING.md)
- [ ] My code follows the project's code style
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the documentation accordingly
- [ ] For `cosh`: Lint passes, type check passes, and tests pass
- [ ] For `sec-core` (Rust): `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] For `sec-core` (Python): Ruff format and pytest pass
- [ ] For `skill`: Skill directory structure is valid and shell scripts pass syntax check
- [ ] For `sight`: `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] For `tokenless`: `cargo clippy -- -D warnings` and `cargo fmt --check` pass
- [ ] Lock files are up to date (`package-lock.json` / `Cargo.lock`)

### CODE REVIEW

File: src/agent-sec-core/tests/e2e/prompt-scanner/e2e_test.py [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+# ---------------------------------------------------------------------------
+# A. Basic functionality
+# ---------------------------------------------------------------------------
+
+
+class TestBasicScan:
+    """Verify fundamental scan-prompt behaviour."""
+
+    def test_empty_text_returns_error(self) -> None:
+        """--text '' produces exit_code=0 but empty stdout (CLI skips empty input)."""
```
Comment (edonyzpc) [inline src/agent-sec-core/tests/e2e/prompt-scanner/e2e_test.py:90]: `prompt_scanner/cli.py` 对对输入的处理逻辑：
- `--text` 传入空字符串时，text 为 ""（非 None）；
- 无 `--text` 会打印 "Error: Ninput received from stdin." 并 Exit(code=1)

e2e 测试显式断言 stdout 为空（“fail-open on empty input” 且“静默无输出”），与现有 CLI 行为相矛盾。确认一下测试用例运行是否会失败

**Review by edonyzpc — CHANGES_REQUESTED** [review]

**Review by edonyzpc — APPROVED** [review]
LGTM
